src/server/zset_family.cc

```diff
@@ -3032,7 +3079,49 @@ void ZSetFamily::GeoSearch(CmdArgList args, ConnectionContext* cntx) {
   if (!by_set) {
     return cntx->SendError(kSyntaxErr);
   }
-  GeoSearchGeneric(cntx, shape, key, geo_ops);
+  // parsing completed
+
+  // verify key
+  auto* rb = static_cast<RedisReplyBuilder*>(cntx->reply_builder());
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    return OpKeyExisted(t->GetOpArgs(shard), key);
+  };
+  OpResult<void> result = cntx->transaction->ScheduleSingleHopT(std::move(cb));
```

Comment:
I do not think the implementation of this function is correct. You use ScheduleSingleHop method on the transaction but this method is intended for operations that require only a single hop. Here you call it multiple times - which breaks the atomicity of the whole command and probably incorrect in general.   The correct way is to use Schedule() once, and then `Execute(..., false)` one or more times and finalize with Execute(...., true) or `Conclude` method in case the transaction finishes prematurely.